pythran/pythonic/types/list.hpp

```diff
@@ -148,63 +148,52 @@ namespace pythonic {
                 static const size_t value = utils::nested_container_depth<list>::value;
                 static const bool is_vectorizable = true;
                 static const bool is_strided = false;
-                struct fake_shape {
-                  list const & the_list;
-                  fake_shape(list const & the_list) : the_list{the_list}{}
-                  template<class E> void init_shape(array<long, value>& res, E const& e, utils::int_<1>) const {
-                    res[value - 1] = e.size();
-                  }
-                  template<class E, size_t L> void init_shape(array<long, value>& res, E const& e, utils::int_<L>) const {
-                    res[value - L] = e.size();
-                    init_shape(res, e[0], utils::int_<L-1>{});
-                  }
-                  operator array<long, value>() const {
-                    array<long, value> res;
-                    init_shape(res, the_list, utils::int_<value>{});
-                    return res;
-                  }
-                } shape;
-
-
```

Comment:
> I already noticed this problem but choose to ignore it (for now).
> 
> I did this modification for 3 reasons:
> - To have a common interface for future arrayLike Interface
> - This class doesn't work on Ubuntu (:-s because Ubuntu still use gcc 4.8.x which generate wrong code and create an infinite loop)
> - It happend that calling the shape argument of a list failed (because of missing size() function for some arguments)
> ex: `np.sum([np.array([1, 2]), np.array([3, 4])])`

ok for size(), but couldn't you rewrite the class so that it works on
gcc 4.8 instead of introducing a new issue?